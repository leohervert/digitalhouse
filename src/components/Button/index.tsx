import React from 'react';
import {TouchableOpacity} from 'react-native';
import type {TouchableOpacityProps} from 'react-native';

export const Button = ({
  style,
  onPress,
  children,
}: TouchableOpacityProps): JSX.Element => {
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      {children}
    </TouchableOpacity>
  );
};
