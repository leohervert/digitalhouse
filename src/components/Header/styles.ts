import {StyleSheet} from 'react-native';
import {fontStyles} from '../../styles/AppStyles';

export const styles = StyleSheet.create({
  textBold: {
    ...fontStyles.defaultFont,
    paddingTop: 20,
    fontSize: 20,
    lineHeight: 27.32,
  },
  text: {
    ...fontStyles.lightFont,
    fontSize: 16,
    lineHeight: 21.86,
  },
});
