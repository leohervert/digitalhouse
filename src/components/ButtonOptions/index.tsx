import React, {useContext} from 'react';
import {View, Text} from 'react-native';
import {ProductContext} from '../../context/ProductContext';
import {Button} from '../Button';
import {styles} from './styles';

export const ButtonOptions = () => {
  const {toggleDisplay, productState} = useContext(ProductContext);
  const {isRedemption} = productState;
  const handleOnPress = (value: null | boolean) => {
    toggleDisplay(value);
  };

  return (
    <View style={styles.container}>
      {isRedemption !== null && (
        <Button
          testID="allButton"
          onPress={() => handleOnPress(null)}
          style={styles.mainButton}>
          <Text testID="allText" style={styles.textButton}>
            Todos
          </Text>
        </Button>
      )}
      {isRedemption === null && (
        <View style={styles.buttonContainer}>
          <Button
            testID="savedButton"
            onPress={() => handleOnPress(false)}
            style={[styles.rightOption, styles.subButton]}>
            <Text testID="savedText" style={styles.textButton}>
              Ganados
            </Text>
          </Button>
          <Button
            testID="redeemedButton"
            onPress={() => handleOnPress(true)}
            style={[styles.leftOption, styles.subButton]}>
            <Text testID="redeemedText" style={styles.textButton}>
              Canjeados
            </Text>
          </Button>
        </View>
      )}
    </View>
  );
};
