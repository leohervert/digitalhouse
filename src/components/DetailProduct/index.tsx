import {format} from 'date-fns';
import React from 'react';
import {View, Text} from 'react-native';
import {styles} from './styles';

export const DetailProduct = ({
  points,
  is_redemption,
  createdAt,
}: ProductModel) => {
  return (
    <View style={styles.container}>
      <Text style={styles.detailTitle}>Detalles del producto: </Text>
      <Text style={[styles.detailText, styles.dateText]}>
        Comprando el {format(new Date(createdAt), 'dd/MM/yyyy')}
      </Text>
      {is_redemption ? (
        <Text style={styles.detailTitle}>Con esta compra usaste: </Text>
      ) : (
        <Text style={styles.detailTitle}>Con esta compra acumulaste: </Text>
      )}
      <Text style={[styles.detailText, styles.pointsText]}>
        {points} puntos
      </Text>
    </View>
  );
};
