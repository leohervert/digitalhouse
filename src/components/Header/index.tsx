import React from 'react';
import {Text, View} from 'react-native';
import {styles} from './styles';
export const Header = () => {
  return (
    <View>
      <Text testID="HeaderTitle" style={styles.textBold}>
        Bienvenido de vuelta!
      </Text>
      <Text testID="UserTitle" style={styles.text}>
        Ruben Rodriguez
      </Text>
    </View>
  );
};
