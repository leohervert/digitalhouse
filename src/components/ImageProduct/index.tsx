import React from 'react';
import {View, Image} from 'react-native';
import {styles} from './styles';

type Props = {
  image: string;
};
export const ImageProduct = ({image}: Props): JSX.Element => {
  return (
    <View style={styles.container}>
      <Image style={styles.image} source={{uri: image}} />
    </View>
  );
};
