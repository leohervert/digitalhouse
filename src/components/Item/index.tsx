import React from 'react';
import {View, Text, Image, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {format} from 'date-fns';
import {useNavigation} from '@react-navigation/native';

type Props = ProductModel;

export const Item = ({
  product,
  createdAt,
  image,
  points,
  is_redemption,
}: Props): JSX.Element => {
  const currentDate = new Date(createdAt);
  const navigation = useNavigation();
  const handleChangeScreen = () => {
    navigation.navigate('Product', {
      product: product,
      createdAt: createdAt,
      image: image,
      points: points,
      is_redemption: is_redemption,
    });
  };
  return (
    <TouchableOpacity onPress={handleChangeScreen} style={styles.touchable}>
      <View style={styles.container}>
        <Image style={styles.tinyImage} source={{uri: image}} />
        <View style={styles.titleContainer}>
          <Text style={styles.nameText}>{product}</Text>
          <Text style={styles.dateText}>
            {format(currentDate, 'dd/MM/yyyy')}
          </Text>
        </View>
        <View style={styles.pointsContainer}>
          <Text style={styles.pointsText}>
            <Text style={is_redemption ? styles.redColor : styles.greenColor}>
              {is_redemption ? '-' : '+'}
            </Text>
            {points} {' >'}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};
