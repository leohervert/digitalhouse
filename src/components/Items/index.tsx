import React, {useContext, useState, useEffect} from 'react';
import {ActivityIndicator, View, ScrollView} from 'react-native';
import {ProductContext} from '../../context/ProductContext';
import {Item} from '../Item';
import {styles} from './styles';

const ItemsComponent = () => {
  const {productState} = useContext(ProductContext);
  const {products, loading, isRedemption} = productState;
  const [state, setState] = useState<ProductModel[]>([]);
  useEffect(() => {
    if (products.length > 0) {
      setState(
        isRedemption === null
          ? products
          : isRedemption === true
          ? products.filter(x => x.is_redemption)
          : products.filter(x => !x.is_redemption),
      );
    }
  }, [products, isRedemption]);

  return (
    <View style={styles.container}>
      <ScrollView>
        {loading && <ActivityIndicator size="large" animating={loading} />}
        {!loading &&
          state.map((item: ProductModel) => <Item key={item.id} {...item} />)}
      </ScrollView>
    </View>
  );
};

export const Items = React.memo(ItemsComponent);
