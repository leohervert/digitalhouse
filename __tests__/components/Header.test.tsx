import React from 'react';
import {create} from 'react-test-renderer';
import {Header} from '../../src/components/Header';

const tree = create(<Header />);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
