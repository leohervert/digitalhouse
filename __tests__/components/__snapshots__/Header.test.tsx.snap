// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View>
  <Text
    style={
      {
        "color": "#020202",
        "fontFamily": "sans-serif",
        "fontSize": 20,
        "fontStyle": "normal",
        "fontWeight": "800",
        "lineHeight": 27.32,
        "paddingTop": 20,
      }
    }
    testID="HeaderTitle"
  >
    Bienvenido de vuelta!
  </Text>
  <Text
    style={
      {
        "color": "#020202",
        "fontFamily": "sans-serif",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 21.86,
      }
    }
    testID="UserTitle"
  >
    Ruben Rodriguez
  </Text>
</View>
`;
